src/components/query/QueryDirective.js

```diff
@@ -254,22 +255,37 @@
       var features = [];
       var params = getParamsByLayer($scope.filters);
       if (params.length == 0) {
+        if ($scope.useBbox) {
+          $scope.searchByGeometry();
+        }
         $scope.options.features = [];
         $scope.loading = false;
         return;
       }
-      var lang = $translate.use();
+
+      if (!$scope.useBbox) {
+        $scope.geometry = null;
+      }
+
+      var common = {
```

Comment:
We could move 270-274 to 265 and then merge both if-statements on 266 and 275. I think it would increase readability.